javascript/react/redux-tips/avoid-array-mutations.md

```diff
@@ -28,35 +29,99 @@ aspects:
 ---
 ## Content
 
-There are different ways of creating a **pure** function needed in **Redux** for generating a new array of items.
+There are different ways of creating a **pure** function needed in **Redux** for generating a new array of items. These types of functions are necessary because Redux takes a given state, passes it to each `reducer`, and finally it expects a new object if there are any changes. If we mutate the old state in a `reducer`, both the old state and the new state point to the same object, resulting in Redux thinking that there has been no change.
```

Comment:
A lot of our content talks about "pure function" but doesn't explain it (unless I'm missing something somewhere else in the content).

This paragraph tries to explain it but feels confusing.

In fact, Redux doesn't need "pure functions", it needs "pure reducer functions" only. All other functions don't have to be pure. 

[Useful content to get inspiration from](https://redux.js.org/recipes/structuring-reducers/prerequisite-concepts#immutable-data-management)

Reply:
In the previous workout `redux-i` we mention that `reducers` must be a pure functions (in the reducers insight), and we have a separate insight dedicated to explaining what a pure function is. I've changed the phrase to:

These types of functions are necessary for `reducers` because Redux takes a...